internal/kibana/security_exception_item/models.go

```diff
@@ -0,0 +1,994 @@
+		req.ItemId = &itemID
+	}
+
+	// Set optional namespace_type
+	if utils.IsKnown(m.NamespaceType) {
+		nsType := kbapi.SecurityExceptionsAPIExceptionNamespaceType(m.NamespaceType.ValueString())
+		req.NamespaceType = &nsType
+	}
+
+	// Set optional os_types
+	if utils.IsKnown(m.OsTypes) {
+		osTypes := utils.SetTypeAs[string](ctx, m.OsTypes, path.Empty(), &diags)
+		if diags.HasError() {
+			return nil, diags
+		}
+		if len(osTypes) > 0 {
+			osTypesArray := make(kbapi.SecurityExceptionsAPIExceptionListItemOsTypeArray, len(osTypes))
+			for i, osType := range osTypes {
+				osTypesArray[i] = kbapi.SecurityExceptionsAPIExceptionListOsType(osType)
+			}
+			req.OsTypes = &osTypesArray
+		}
+	}
+
+	// Set optional tags
+	if utils.IsKnown(m.Tags) {
+		tags := utils.SetTypeAs[string](ctx, m.Tags, path.Empty(), &diags)
+		if diags.HasError() {
+			return nil, diags
+		}
+		if len(tags) > 0 {
+			tagsArray := kbapi.SecurityExceptionsAPIExceptionListItemTags(tags)
+			req.Tags = &tagsArray
+		}
+	}
+
+	// Set optional meta
+	if utils.IsKnown(m.Meta) {
+		var meta kbapi.SecurityExceptionsAPIExceptionListItemMeta
+		unmarshalDiags := m.Meta.Unmarshal(&meta)
+		diags.Append(unmarshalDiags...)
+		if diags.HasError() {
+			return nil, diags
+		}
+		req.Meta = &meta
+	}
+
+	// Set optional comments
+	if utils.IsKnown(m.Comments) {
+		comments := utils.ListTypeAs[CommentModel](ctx, m.Comments, path.Empty(), &diags)
+		if diags.HasError() {
+			return nil, diags
+		}
+		if len(comments) > 0 {
+			commentsArray := make(kbapi.SecurityExceptionsAPICreateExceptionListItemCommentArray, len(comments))
+			for i, comment := range comments {
+				commentsArray[i] = kbapi.SecurityExceptionsAPICreateExceptionListItemComment{
+					Comment: kbapi.SecurityExceptionsAPINonEmptyString(comment.Comment.ValueString()),
+				}
+			}
+			req.Comments = &commentsArray
+		}
+	}
+
+	// Set optional expire_time
+	if utils.IsKnown(m.ExpireTime) {
+		// Check version support for expire_time
+		if supported, versionDiags := client.EnforceMinVersion(ctx, MinVersionExpireTime); versionDiags.HasError() {
+			diags.Append(diagutil.FrameworkDiagsFromSDK(versionDiags)...)
+			return nil, diags
+		} else if !supported {
+			diags.AddError("expire_time is unsupported",
+				fmt.Sprintf("expire_time requires server version %s or higher", MinVersionExpireTime.String()))
+			return nil, diags
+		}
+
+		expireTime, d := m.ExpireTime.ValueRFC3339Time()
+		diags.Append(d...)
+		if diags.HasError() {
+			return nil, diags
+		}
+
+		expireTimeAPI := kbapi.SecurityExceptionsAPIExceptionListItemExpireTime(expireTime.Format("2006-01-02T15:04:05.000Z"))
+		req.ExpireTime = &expireTimeAPI
+	}
+
+	return req, diags
+}
+
+// toUpdateRequest converts the Terraform model to API update request
+func (m *ExceptionItemModel) toUpdateRequest(ctx context.Context, resourceId string, client clients.MinVersionEnforceable) (*kbapi.UpdateExceptionListItemJSONRequestBody, diag.Diagnostics) {
```

Comment:
There's a lot of duplication between `toCreate` and `toUpdate`. I wonder if we can just build the update request from the create request, or reuse some of the common code.